org.eclipse.search/search/org/eclipse/search/internal/ui/text/ReplaceRefactoring.java

```diff
@@ -290,31 +290,36 @@ public boolean hasMatches() {
 	 * @since 3.7
 	 */
 	private boolean isMatchToBeIncluded(FileMatch match) {
-		IFile file= match.getFile();
-		URI uri= file.getLocationURI();
+		IFile file = match.getFile();
+		URI uri = file.getLocationURI();
 		if (uri == null) {
 			return true;
 		}
 		if (file.equals(fAlreadyCollected.get(uri))) {
 			return true; // another FileMatch for an IFile which already had
 							// matches
 		}
-
-		for (URI uri2 : fAlreadyCollected.keySet()) {
-			if (URIUtil.equals(uri2, uri)) {
-				if (fIgnoredMatches == null) {
-					fIgnoredMatches= new HashMap<>();
-				}
-				ArrayList<FileMatch> matches= fIgnoredMatches.get(uri);
-				if (matches == null) {
-					matches= new ArrayList<>();
-					fIgnoredMatches.put(uri, matches);
-				}
-				matches.add(match);
-				return false;
+		Object key; // IFileStore or URI, see URIUtil.equals(URI, URI)
+		try {
+			key = EFS.getStore(uri);
+		} catch (CoreException e) {
+			// fall back to default equality test
+			key = uri;
+		}
+		if (fAlreadyCollected.containsKey(key)) {
```

Comment:
But that doesn't match what URIUtil does - it converts arguments before doing equals to `EFS.getStore(one)`. Here only the key is converted to IFileStore or URI, but so we compare apples with oranges?

Reply:
see line 304: i convert too. the map then stores both apples and oranges in the bin (line 322+323). So i can test if either apple or orange is inside.

Reply:
Arrgh, I've overlooked that. 
Please add comment to the map definition that keys can be both apples and oranges.

Reply:
Beside this, LGTM

Reply:
done